students/tammyd/lesson03/mailroom_pt1.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+
+"""
+
+Mailroom program for a non profit
+
+"""
+
+import sys
+
+donor_db = [("Britney Spears", [50.99, 100.99, 2000.99]),
+            ("Christina Aguilera", [50.99, 100.99, 2000.99]),
+            ("Mandy Moore", [50.99, 100.99, 2000.99]),
+            ("Jessica Simpson", [50.99, 100.99, 2000.99]),
+            ("Queen Bey", [50.99, 100.99, 2000.99]),
+            ]
+
+def mainloop():
+    print("Welcome to the Mailroom \n")
+
+def main_menu():
+    print("Main Menu:")
+    print(
+      "T - Send a thank you\n"
+      "R - Create a report\n"
+      "Q - Quit the program"
+      )
+    menu_selection = input("\nChoose a letter:  ")
+    if menu_selection.upper().strip() == "T":
+        thank_you()
+    elif menu_selection.upper().strip() == "R":
+        create_report()
+    elif menu_selection.upper().strip() == "Q":
+        quit_program()
+    else:
+        print("Invalid response -- select T, R, or Q \n")
+        return main_menu()
+
+def thank_you():
+    print("\nMenu: Send a Thank You")
+    while True:
+        donor_name = input("Enter the donor's full name. \nEnter 'list' to see a list of donors, \n or 'menu' to return to the main menu: ").strip()
+        if donor_name.lower() == "list":
+            print("\nDonor database:")
+            for donor in donor_db:
+                print(donor[0])
+            print("\n")
+        elif donor_name.lower() == "menu":
+            return
+        else:
+            break
+
+    while True:
+        donor_amount = input("Enter amount of donation amount (or 'menu' to exit): ").strip()
+        if donor_amount.lower() == "menu":
+            return main_menu()
+        else:
+            break
+    #I borrowed this from the example solution. Could you please explain None to me? 
+    donor = find_donor(donor_name)
+    if donor is None:
+        donor = (donor_name, [])
+        donor_db.append(donor_name)
+
+    donor[1].append(donor_amount)
+    
+    print(ty_letter(donor_name))
+
+
+def find_donor(donor_name):
+    for donor in donor_db:
+        if donor_name.strip().lower() == donor[0].lower():
+            return donor
+    return None
+
+
+def ty_letter(donor_name):
+    print("\nMenu: Send a Thank You, Letter")
+    #How can I input the donation amount?? Why does donor_amount not work here?
```

Comment:
donor_amount only exists in thank_you(), so this function,  ty_letter(), doesn't see it. You can pass it in as an argument here if you add a parameter for donor_amount. i.e. change your function definition to:
```
def ty_letter(donor_name, donor_amount):
```